t/lib/WTSI/NPG/HTS/TreePublisherTest.pm

```diff
@@ -107,6 +107,44 @@ sub publish_tree : Test(58) {
   check_metadata($irods, map { catfile($irods_tmp_coll, $_) } @observed_paths);
 }
 
+sub npg_publish_tree_pl_metadata_from_stdin : Test(3) {
+  my $source_path = "${data_path}/treepublisher";
+  my @attributes;  
+  push @attributes, {attribute => 'attr1', value => 'val1', units => q[]};
+  push @attributes, {attribute => 'attr2', value => 'val2', units => q[]};
```

Comment:
This is correct, but this is better built explicitly because it saves the reader from constructing
the input mentally, when they read it:

 e.g.:

```
my @attributes = (
    {attribute => 'attr1', value => 'val1', units => q[]},
    {attribute => 'attr2', value => 'val2', units => q[]},
);
```